x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/EstimateMemoryUsageAction.java

```diff
@@ -71,11 +72,11 @@ public Response(StreamInput in) throws IOException {
         }
 
         public ByteSizeValue getExpectedMemoryWithoutDisk() {
-            return expectedMemoryWithoutDisk;
+            return expectedMemoryWithoutDisk != null ? max(expectedMemoryWithoutDisk, MIN_MODEL_MEMORY_LIMIT) : null;
```

Comment:
May I suggest to apply the limit in the constructor instead? It will be clearer to do it there instead of relying on the getter to apply the correction.

Reply:
Good suggestion!

Done.